packages/core/strapi/lib/core-api/service/pagination.js

```diff
@@ -18,12 +18,12 @@ const getLimitConfigDefaults = () => ({
  * @param {number?} maxLimit - maxlimit used has capping
  * @returns {number}
  */
-const applyMaxLimit = (limit, maxLimit) => {
-  if (maxLimit && (limit === -1 || limit > maxLimit)) {
-    return maxLimit;
+const applyMaxLimit = (limit, maxLimit = null) => {
```

Comment:
The Math.max doesn't really make sense in the applyMaxLimit function. You should keep this out of this function